app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -237,100 +246,103 @@ export function buildResolver(baseName) {
-
-      if (decamelized === "javascripts/admin") {
+      if (parsedName.fullNameWithoutType === "admin") {
         return Ember.TEMPLATES["admin/templates/admin"];
       }
 
-      if (
-        decamelized.indexOf("admin") === 0 ||
-        decamelized.indexOf("javascripts/admin") === 0
+      let namespaced, match;
+
+      if (parsedName.fullNameWithoutType.startsWith("components/")) {
+        // Look up components as-is
+        namespaced = parsedName.fullNameWithoutType;
+      } else if (/^admin[_\.-]/.test(parsedName.fullNameWithoutType)) {
+        namespaced = parsedName.fullNameWithoutType.slice(6);
+      } else if (
+        (match = parsedName.fullNameWithoutType.match(/^admin([A-Z])(.+)$/))
       ) {
-        decamelized = decamelized.replace(/^admin\_/, "admin/templates/");
-        decamelized = decamelized.replace(/^admin\./, "admin/templates/");
-        decamelized = decamelized.replace(/\./g, "_");
-
-        const dashed = decamelized.replace(/_/g, "-");
-        return (
-          Ember.TEMPLATES[decamelized] ||
-          Ember.TEMPLATES[dashed] ||
-          Ember.TEMPLATES[dashed.replace("admin-", "admin/")]
+        namespaced = `${match[1].toLowerCase()}${match[2]}`;
+      }
+
+      if (namespaced) {
+        let adminParsedName = this.parseName(
+          `template:admin/templates/${namespaced}`
         );
+        return this.findTemplate(adminParsedName);
       }
-    },
+    }
 
     findWizardTemplate(parsedName) {
-      let decamelized = decamelize(parsedName.fullNameWithoutType);
-      if (decamelized.startsWith("components")) {
-        let comPath = `wizard/templates/${decamelized}`;
-        const compTemplate =
-          Ember.TEMPLATES[`javascripts/${comPath}`] || Ember.TEMPLATES[comPath];
-        if (compTemplate) {
-          return compTemplate;
-        }
+      // let decamelized = decamelize(parsedName.fullNameWithoutType);
+      // if (decamelized.startsWith("components")) {
+      //   let comPath = `wizard/templates/${decamelized}`;
+      //   const compTemplate =
+      //     Ember.TEMPLATES[`javascripts/${comPath}`] || Ember.TEMPLATES[comPath];
+      //   if (compTemplate) {
+      //     return compTemplate;
+      //   }
+      // }
+
+      // if (decamelized === "javascripts/wizard") {
+      //   return Ember.TEMPLATES["wizard/templates/wizard"];
+      // }
+
+      // if (
+      //   decamelized.startsWith("wizard") ||
+      //   decamelized.startsWith("javascripts/wizard")
+      // ) {
+      //   decamelized = decamelized.replace(/^wizard\_/, "wizard/templates/");
+      //   decamelized = decamelized.replace(/^wizard\./, "wizard/templates/");
+      //   decamelized = decamelized.replace(/\./g, "_");
+
+      //   const dashed = decamelized.replace(/_/g, "-");
+      //   return (
+      //     Ember.TEMPLATES[decamelized] ||
+      //     Ember.TEMPLATES[dashed] ||
+      //     Ember.TEMPLATES[dashed.replace("wizard-", "wizard/")]
+      //   );
+      // }
```

Comment:
Can we lose the commented stuff, or is this a pending TODO?

Reply:
Yes, we can lose this. Was part of a rebase I forgot to take out.